### PR TITLE
fixes all failing tests on develop branch.

### PR DESCRIPTION
**Pull request by ShurakaiSoft**
fixed all failing tests
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/agco/elastic-harvesterjs/pull/89%23issuecomment-266112033%22%2C%20%22https%3A//github.com/agco/elastic-harvesterjs/pull/89%23issuecomment-266520301%22%2C%20%22https%3A//github.com/agco/elastic-harvesterjs/pull/89%23discussion_r92110137%22%2C%20%22https%3A//github.com/agco/elastic-harvesterjs/pull/89%23issuecomment-266678138%22%2C%20%22https%3A//github.com/agco/elastic-harvesterjs/pull/89%23issuecomment-266757762%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/agco/elastic-harvesterjs/pull/89%23issuecomment-266112033%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Weird%2C%20different%20tests%20failing%20with%20each%20test%20run...%22%2C%20%22created_at%22%3A%20%222016-12-09T20%3A15%3A01Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1182007%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/ShurakaiSoft%22%7D%7D%2C%20%7B%22body%22%3A%20%22%5Cn%5B%21%5BCoverage%20Status%5D%28https%3A//coveralls.io/builds/9242605/badge%29%5D%28https%3A//coveralls.io/builds/9242605%29%5Cn%5CnCoverage%20remained%20the%20same%20at%2077.778%25%20when%20pulling%20%2A%2A0b35aa5159f5e0cad640bc838d47f59fcd1a9957%20on%20fix-broken-tests%2A%2A%20into%20%2A%2A91cc0ea5adf51908d59b191507caa57e3356e9ff%20on%20develop%2A%2A.%5Cn%22%2C%20%22created_at%22%3A%20%222016-12-12T19%3A03%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/2354108%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/coveralls%22%7D%7D%2C%20%7B%22body%22%3A%20%22%5C%22deletes%20Should%20correctly%20sync%20primary%20resource%20deletes%5C%22%20and%20%5Cr%5Cn%5C%22deletes%20Should%20correctly%20sync%20resources%20with%20invalid%20data%20graphs%5C%22%20are%20failing%20for%20me.%22%2C%20%22created_at%22%3A%20%222016-12-13T08%3A37%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/431064%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/blabno%22%7D%7D%2C%20%7B%22body%22%3A%20%22%5Cn%5B%21%5BCoverage%20Status%5D%28https%3A//coveralls.io/builds/9258222/badge%29%5D%28https%3A//coveralls.io/builds/9258222%29%5Cn%5CnCoverage%20remained%20the%20same%20at%2077.778%25%20when%20pulling%20%2A%2A1a0741d349289f8dd80c2503e0ee2cf0b1f02e13%20on%20fix-broken-tests%2A%2A%20into%20%2A%2A91cc0ea5adf51908d59b191507caa57e3356e9ff%20on%20develop%2A%2A.%5Cn%22%2C%20%22created_at%22%3A%20%222016-12-13T14%3A50%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/2354108%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/coveralls%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%200b35aa5159f5e0cad640bc838d47f59fcd1a9957%20test/customRouting.spec.js%2019%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/agco/elastic-harvesterjs/pull/89%23discussion_r92110137%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Promise.map%20is%20shorter.%22%2C%20%22created_at%22%3A%20%222016-12-13T07%3A21%3A57Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/431064%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/blabno%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/customRouting.spec.js%3AL94-103%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/agco/elastic-harvesterjs/pull/89#issuecomment-266112033'>General Comment</a></b>
- <a href='https://github.com/ShurakaiSoft'></a> Weird, different tests failing with each test run...
- <a href='https://github.com/coveralls'></a> [Coverage Status](https://coveralls.io/builds/9242605)
Coverage remained the same at 77.778% when pulling **0b35aa5159f5e0cad640bc838d47f59fcd1a9957 on fix-broken-tests** into **91cc0ea5adf51908d59b191507caa57e3356e9ff on develop**.
- <a href='https://github.com/blabno'></a> "deletes Should correctly sync primary resource deletes" and
"deletes Should correctly sync resources with invalid data graphs" are failing for me.
- <a href='https://github.com/coveralls'></a> [Coverage Status](https://coveralls.io/builds/9258222)
Coverage remained the same at 77.778% when pulling **1a0741d349289f8dd80c2503e0ee2cf0b1f02e13 on fix-broken-tests** into **91cc0ea5adf51908d59b191507caa57e3356e9ff on develop**.
- [ ] <a href='#crh-comment-Pull 0b35aa5159f5e0cad640bc838d47f59fcd1a9957 test/customRouting.spec.js 19'></a> &nbsp;<b><a href='https://github.com/agco/elastic-harvesterjs/pull/89#discussion_r92110137'>File: test/customRouting.spec.js:L94-103</a></b>
- <a href='https://github.com/blabno'></a> Promise.map is shorter.

<a href='https://www.codereviewhub.com/agco/elastic-harvesterjs/pull/89?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/agco/elastic-harvesterjs/pull/89?approve=1'></a>&nbsp;<a href='https://github.com/agco/elastic-harvesterjs/pull/89'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by ssebro** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/agco/elastic-harvesterjs/pull/89'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by ShurakaiSoft** [comment]
Weird, different tests failing with each test run...

File: test/customRouting.spec.js [hunk 1]
```diff
@@ -93,6 +94,9 @@ describe('Custom Routing', function () {
                 error: false
             }
         }
+        return Promise.all(_.forEach(config.harvester.options.es_types, function (indexName) {
```
Comment (blabno) [inline test/customRouting.spec.js]: Promise.map is shorter.

File: test/global.spec.js [hunk 2]
```diff
@@ -1,12 +1,24 @@
...
 
 before(function () {
     this.timeout(config.esIndexWaitTime + 1000);
     return harvesterApp.apply(this).then(function (harvesterInstance) {
         return events(harvesterInstance);
     });
+
 });
+
+// beforeEach(function () {
```
Comment (blabno) [inline test/global.spec.js]: I think that we can revert all changes on this file.

**Review by blabno — CHANGES_REQUESTED** [review]
For me the tests are constantly failing. Especially the deletes.

**Comment by blabno** [comment]
"deletes Should correctly sync primary resource deletes" and 
"deletes Should correctly sync resources with invalid data graphs" are failing for me.

**Pull request merged**